components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/jwt_auth_provider.bal

```diff
@@ -97,11 +107,41 @@ public type JwtAuthProvider object {
                         } else {
                             printDebug(KEY_JWT_AUTH_PROVIDER, "jti claim not found in the jwt");
                         }
+                        if(self.className != "" || (claimsSet is map<anydata>[] && claimsSet.length() > 0)) {
+                            var jwtTokenClaimCached = self.gatewayCache.retrieveClaimMappingCache(jwtToken);
+                            if (jwtTokenClaimCached is runtime:Principal) {
+                                invocationContext.principal =  jwtTokenClaimCached;
+                                printDebug(KEY_JWT_AUTH_PROVIDER, "Moddified claims in the cache");
+                            } else {
+                                printDebug(KEY_JWT_AUTH_PROVIDER, "Moddified claims is not in the cache");
+                                var result = doMappingContext(invocationContext, self.className, self.claims,
+                                    self.classLoaded, jwtPayloadFromCache, self.jwtValidatorConfig, self.gatewayCache, authContext);
+                                if (result is auth:Error){
+                                    return result;
+                                }
+                                jwtToken = authContext?.authToken.toString();
```

Comment:
jwtToken is already retrieved from above lines. line 62. Why retrieve it again?

Reply:
Here we have to take modified jwt token with claims from auth context.